src/vulkan_pnext_chain.cpp

```diff
@@ -0,0 +1,188 @@
+// Unit test to try out various combinations of GPU memory copying and synchronization with CPU-side
+// memory mapping.
+
+#include "vulkan_common.h"
+
+static vulkan_req_t reqs;
+
+static void show_usage()
+{
+}
+
+static bool test_cmdopt(int& i, int argc, char** argv, vulkan_req_t& reqs)
+{
+	return false;
+}
+
+int main(int argc, char** argv)
+{
+	reqs.usage = show_usage;
+	reqs.cmdopt = test_cmdopt;
+	reqs.instance_extensions = { "VK_EXT_debug_utils" };
+	vulkan_setup_t vulkan = test_init(argc, argv, "vulkan_pnext_chain", reqs);
+	VkResult result;
+
+	// Create everything to submit debug labels to mark end of frames
+
+	VkQueue queue;
+	vkGetDeviceQueue(vulkan.device, 0, 0, &queue);
+
+	VkCommandPoolCreateInfo commandPoolCreateInfo;
+	commandPoolCreateInfo.sType = VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO;
+	commandPoolCreateInfo.pNext = nullptr;
+	commandPoolCreateInfo.queueFamilyIndex = 0;
+	commandPoolCreateInfo.flags = 0;
+
+	VkCommandPool commandPool;
+	result = vkCreateCommandPool(vulkan.device, &commandPoolCreateInfo, nullptr, &commandPool);
+	assert(result == VK_SUCCESS);
+
+	VkCommandBufferAllocateInfo commandBufferAllocateInfo;
+	commandBufferAllocateInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+	commandBufferAllocateInfo.pNext = nullptr;
+	commandBufferAllocateInfo.commandPool = commandPool;
+	commandBufferAllocateInfo.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
+	commandBufferAllocateInfo.commandBufferCount = 1;
+
+	VkCommandBuffer commandBuffer;
+	result = vkAllocateCommandBuffers(vulkan.device, &commandBufferAllocateInfo, &commandBuffer);
+	assert(result == VK_SUCCESS);
+
+	VkCommandBufferBeginInfo commandBufferBeginInfo;
+	commandBufferBeginInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
+	commandBufferBeginInfo.pNext = nullptr;
+	commandBufferBeginInfo.flags = 0;
+	commandBufferBeginInfo.pInheritanceInfo = nullptr;
+	
+	result = vkBeginCommandBuffer(commandBuffer, &commandBufferBeginInfo);
+	assert(result == VK_SUCCESS);
+
+	VkDebugUtilsLabelEXT debugLabel;
+	debugLabel.sType = VK_STRUCTURE_TYPE_DEBUG_UTILS_LABEL_EXT;
+	debugLabel.pNext = nullptr;
+	debugLabel.pLabelName = "vr-marker,frame_end,type,application";
+	debugLabel.color[0] = 1.f;
+	debugLabel.color[1] = 0.f;
+	debugLabel.color[2] = 0.f;
+	debugLabel.color[3] = 1.f;
+
+	PFN_vkCmdInsertDebugUtilsLabelEXT pfnVkCmdInsertDebugUtilsLabelEXT = (PFN_vkCmdInsertDebugUtilsLabelEXT) vkGetInstanceProcAddr(vulkan.instance, "vkCmdInsertDebugUtilsLabelEXT");
```

Comment:
Can you add it to the vulkan object, see src/vulkan_common.cpp line 441 ?

Reply:
Done.